### ISSUE TITLE
Override default Stackdriver monitoring labels

### ISSUE THREAD

**Issue by mcnamam** [census-instrumentation/opencensus-python#609]
I am trying to override the default labels published to Stackdriver. 
In StackdriverStatsExporter, the exporter is initialized with Options including ``defaut_monitoring_labels``. The comments for this option say: ``
If unset, this defaults to a single label with key "opencensus_task" and value "py-<pid>@<hostname>".``
Later on in the function ``new_label_descriptors``, it appears that it always adds in the ``opencensus_task`` info to the label descriptors even if ``defaut_monitoring_labels`` is set:
```python
def new_label_descriptors(defaults, keys):
    """ create labels for the metric_descriptor
        that will be sent to Stackdriver Monitoring
    """
    label_descriptors = []
    for key, lbl in defaults.items():
        label = {}
        label["key"] = sanitize_label(key)
        label["description"] = lbl
        label_descriptors.append(label)

    for label_key in keys:
        label = {}
        label["key"] = sanitize_label(label_key.key)
        label["description"] = sanitize_label(label_key.description)
        label_descriptors.append(label)
    label_descriptors.append({"key": OPENCENSUS_TASK,
                              "description": OPENCENSUS_TASK_DESCRIPTION})
    return label_descriptors
```
This function is called with the defaults set to default_monitoring_labels option:
```python
    default_labels = self.options.default_monitoring_labels
    if default_labels is None:
        default_labels = {}
    desc_labels = new_label_descriptors(default_labels, oc_md.label_keys)
```
Am I misunderstanding this or is this the desired behavior? Overall I am just looking for a way to avoid publishing a label to Stackdriver with potential for very high cardinality

**Comment by mcnamam** [comment]
It was mentioned to me in another thread that the Go client had an issue and subsequent change to support this:
https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/issues/16
https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/commit/a6442a43b756fe965a0ea004ec8aa1d7e894cc46

I noticed the Python comments say something similar to what they added for Go:
@property
    def default_monitoring_labels(self):
...
        Set this to Labels to avoid getting the
        default "opencensus_task" label.
        You should only do this if you know that
        the Resource you set uniquely identifies this Python process.

Not an urgent ask but would be nice to remove an unnecessary column in all of our Stackdriver charts

**Comment by c24t** [comment]
Thanks for opening the issue @mcnamam. Besides ignoring empty labels, it looks like custom labels weren't being set  on exported metrics at all. This should be fixed in #622.
